pkg/clusteragent/admission/mutate/auto_instrumentation.go

```diff
@@ -189,6 +190,48 @@ func injectApmTelemetryConfig(pod *corev1.Pod) {
 	_ = injectEnv(pod, instrumentationInstallIDEnvVar)
 }
 
+// getLibrariesToInjectForApmInstrumentation returns the list of tracing libraries to inject, when APM Instrumentation is enabled
+// - if apm_config.instrumentation.lib_versions set, returns only tracing libraries from apm_config.instrumentation.lib_versions
+// - if language detection is on and can detect the apps' languages, returns only auto-detected languages
+// - otherwise returns all tracing libraries supported by APM Instrumentation
+func getLibrariesToInjectForApmInstrumentation(pod *corev1.Pod, registry string) ([]libInfo, bool) {
```

Comment:
The function looks correct to me, but I think that extracting each library source to its own function might make the code easier to follow.
I'd extract `librariesFromConfigOptions`, `librariesFromAutoDetection`, and `defaultLibraries` or something similar.
What do you think?

Reply:
Done